examples/aiml/recog.scm

```diff
@@ -0,0 +1,63 @@
+
+; Some very crude experiments
+;
+; This is a rough sketch of the idea that pattern recognition
+; is the dual of pattern matching.  There are many things wrong
+; with the below; its just a sketch.
+
+; Two different pseudo-AIML rules:
+;    I * you   --> I * you too
+;    I love *  --> I like * a lot!
+;
+(BindLink
+	(ListLink
+		(ConceptNode "I")
+		(VariableNode "$star")
+		(ConceptNode "you"))
+	(ListLink
+		(ConceptNode "I")
+		(VariableNode "$star")
+		(ConceptNode "you")
+		(ConceptNode "too")))
+
+(BindLink
+	(ListLink
+		(ConceptNode "I")
+		(ConceptNode "love")
+		(VariableNode "$star"))
+	(ListLink
+		(ConceptNode "I")
+		(ConceptNode "like")
+		(VariableNode "$star")
+		(ConceptNode "a")
+		(ConceptNode "lot!")))
+
+(define data
+	;; A pretend "sentence" that is the "input".
+	(PatternLink
+		(ListLink
+			(ConceptNode "I")
+			(ConceptNode "love")
+			(ConceptNode "you"))))
+
+;; Perform the search.
+(cog-recognize data)
+
+;; At this time, the above will return this:
+(SetLink
+   (ListLink
+      (ConceptNode "I")
+      (VariableNode "$star")
+      (ConceptNode "you")
+   )
+   (ListLink
+      (ConceptNode "I")
+      (ConceptNode "love")
+      (VariableNode "$star")
+   )
+)
```

Comment:
Wow, so given an instance, the pattern recognizer will return all patterns in the atomspace that may have matched that instance, right? That definitely useful for the URE. Can it handle partial matches? Like if the **instance** is
```scheme
(ListLink
   (ConceptNode "I")
   (ConceptNode "love")
   (VariableNode "$star")
)
```
would it return
```scheme
(SetLink
   (ListLink
      (ConceptNode "I")
      (VariableNode "$star-1")
      (VariableNode "$star-2")
   )
   ...
)
```
Assuming
```scheme
(ListLink
   (ConceptNode "I")
   (VariableNode "$star-1")
   (VariableNode "$star-2")
)
```
is a pattern in the atomspace (I suppose present in a BindLink or such).